tests/Tests/Unit/Transport/Rest/JsonStreamDecoderTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/*
+ * Copyright 2021 Google LLC
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Google\ApiCore\Tests\Unit\Transport\Rest;
+
+use Google\ApiCore\Tests\Unit\TestTrait;
+use Google\ApiCore\Transport\Rest\JsonStreamDecoder;
+use Google\LongRunning\Operation;
+use Google\Protobuf\Any;
+use Google\Rpc\Status;
+use GuzzleHttp\Psr7;
+use PHPUnit\Framework\TestCase;
+
+
+class JsonStreamDecoderTest extends TestCase
+{
+    use TestTrait;
+
+    /**
+     * @dataProvider buildResponseStreams
+     */
+    public function testJsonStreamDecoder(array $responses, $decodeType, $stream, $readChunkSizeBytes) {
```

Comment:
Bear with me, I am not familiar with PHP.
Where does `testJsonStreamDecoder` invoke?

Reply:
I don't follow the question. This is a unit test and it instantiates a JsonStreamDecoder then calls `decode` which generates response messages as they are decoded.